File: src/HospitalSlave.java
public class HospitalSlave implements HospitalAction{
    public static void action(String action, Person pers){
        switch (action) {
            case "Убежал из больницы":
                System.out.println("Обслуживающий персонал больницы больше не занимается " + pers.name);
                break;
            case "Приказываю принести яблочное пюре":
                System.out.println("Обслуживающий персонал больницы принёс яблочкное пюре для " + pers.name);
                pers.action("Приказываю принести яблочное пюре");
                break;
            case "Я просил грушевого квасу":
                System.out.println("Обслуживающий персонал больницы принёс грушевый квас для " + pers.name);
                pers.action("Я просил грушевого квасу");
                break;
            case "Квас воняет луком":
                System.out.println("Обслуживающий персонал больницы в тильте и хейтит позера " + pers.name);
                break;
            case "Заказываю котлеты из земляники с грибным соусом":
                System.out.println("Обслуживающий персонал больницы считает, что таких котлет не бывает");
                break;
            default:
                System.out.println("Обслуживающий персонал больницы принёс заказ для " + pers.name);
                break;
        }
    }
}

File: lab4/HospitalSlave.kt
object HospitalSlave : HospitalAction {
    class Hyan(var Bname: String){
        fun rave(){
            println("Все нянечки сбились с ног, исполняя капризы $Bname. Они говорили, что у них спокон веку такого больного не было, что это сущее наказание, а не больной, и чтобы он выздоравливал уж поскорее, что ли.")
        }
        fun mindGames(){
            println("Нянечка устала шататься по городу, она возвращается в больницу в надежде, что он уже забыл о своей собаке,")
        }
    }
    fun action(action: String?, pers: Person) {
        when (action) {
            "Убежал из больницы" -> println("Обслуживающий персонал больницы больше не занимается " + pers.name)
            "Приказываю принести яблочное пюре" -> {
                println("Обслуживающий персонал больницы принёс яблочкное пюре для " + pers.name)
                pers.action()
            }
            "Я просил грушевого квасу" -> {
                println("Обслуживающий персонал больницы принёс грушевый квас для " + pers.name)
                pers.action()
            }
            "Квас воняет луком" -> println("Обслуживающий персонал больницы в тильте и хейтит позера " + pers.name)
            "Заказываю котлеты из земляники с грибным соусом" -> println("Обслуживающий персонал больницы считает, что таких котлет не бывает")
            else -> println("Обслуживающий персонал больницы принёс заказ для " + pers.name)
        }
    }
}